data-prepper-plugins/saas-source-plugins/jira-source/src/test/java/org/opensearch/dataprepper/plugins/source/jira/JiraServiceTest.java

```diff
@@ -1,39 +1,260 @@
+        crawlerTaskExecutor = executorServiceProvider.get();
+        issueType = new ArrayList<>();
+        futureList = new ArrayList<>();
+    }
+
+    @AfterEach
+    void tearDown() {
+        executorServiceProvider.terminateExecutor();
+    }
+
+    @Test
+    void testJiraServiceInitialization() throws JsonProcessingException {
+        jiraSourceConfig = createJiraConfiguration(BASIC, issueType);
+        jiraService = new JiraService(restTemplate, jiraSourceConfig, authConfig);
+        assertNotNull(jiraService);
+    }
+
+    @Test
+    public void testGetJiraEntities() throws JsonProcessingException, InterruptedException, TimeoutException {
+        issueType.add("Task");
+        jiraSourceConfig = createJiraConfiguration(BASIC, issueType);
+        jiraService = spy(new JiraService(restTemplate, jiraSourceConfig, authConfig));
+        List<IssueBean> mockIssues = new ArrayList<>();
+        IssueBean issue1 = createIssueBean(false);
+        mockIssues.add(issue1);
+        IssueBean issue2 = createIssueBean(true);
+        mockIssues.add(issue2);
+
+        when(mockSearchResults.getIssues()).thenReturn(mockIssues);
+        when(mockSearchResults.getTotal()).thenReturn(mockIssues.size());
+
+        doReturn(mockSearchResults).when(jiraService).getAllIssues(any(StringBuilder.class), anyInt(), any(JiraSourceConfig.class));
+
+        Instant timestamp = Instant.ofEpochSecond(0);
+        jiraService.getJiraEntities(jiraSourceConfig, timestamp, itemInfoQueue, futureList, crawlerTaskExecutor);
+
+        waitForFutures();
+
+        assertEquals(mockIssues.size(), itemInfoQueue.size());
+    }
+
+    @Test
+    public void buildIssueItemInfoMultipleFutureThreads() throws JsonProcessingException, InterruptedException, TimeoutException {
+        issueType.add("Task");
+        jiraSourceConfig = createJiraConfiguration(BASIC, issueType);
+        jiraService = spy(new JiraService(restTemplate, jiraSourceConfig, authConfig));
+        List<IssueBean> mockIssues = new ArrayList<>();
+        for (int i = 0; i < 50; i++) {
+            IssueBean issue1 = createIssueBean(false);
+            mockIssues.add(issue1);
+        }
+
+        when(mockSearchResults.getIssues()).thenReturn(mockIssues);
+        when(mockSearchResults.getTotal()).thenReturn(100);
+
+        doReturn(mockSearchResults).when(jiraService).getAllIssues(any(StringBuilder.class), anyInt(), any(JiraSourceConfig.class));
+
+        Instant timestamp = Instant.ofEpochSecond(0);
+        jiraService.getJiraEntities(jiraSourceConfig, timestamp, itemInfoQueue, futureList, crawlerTaskExecutor);
+
+        waitForFutures();
+
+        assertEquals(100, itemInfoQueue.size());
+    }
+
+    @Test
+    public void testGetJiraEntitiesException() throws JsonProcessingException {
+        issueType.add("Task");
+        jiraSourceConfig = createJiraConfiguration(BASIC, issueType);
+        jiraService = spy(new JiraService(restTemplate, jiraSourceConfig, authConfig));
```

Comment:
Same for service. Keep the Service instance local to the test.

Reply:
ok